src/common/src/catalog/column.rs

```diff
@@ -304,6 +341,14 @@ impl From<PbColumnDesc> for ColumnDesc {
             .into_iter()
             .map(ColumnDesc::from)
             .collect();
+        // In case we convert a system column into proto and then back to ColumnDesc, we need to restore the system column meta.
+        let system_column = if prost.column_id == RW_TIMESTAMP_COLUMN_ID.get_id()
+            && prost.name == RW_TIMESTAMP_COLUMN_NAME
+        {
+            Some(SystemColumn::RwTimestamp)
+        } else {
+            None
+        };
```

Comment:
We have a unit test 🥵 to testing `tablecatalog` ==  deserialize(serialize(`tablecatalog`)), if it is acceptable to remove this test, I can remove these lines of code.

Reply:
This test is intentional. Because when we receive updated catalogs from meta, and we deserialize them, we must make sure their fields are correct.

Saw the fix from dylan: https://github.com/risingwavelabs/risingwave/pull/19232/commits/fda1ab38efdd1a8c0115d0406e018d77be06e1a8. Looks good to me.